apps/site/assets/ts/shuttles/components/ShuttlesPage.tsx

```diff
@@ -1,21 +1,115 @@
-import React, { ReactElement } from "react";
-import ShuttlesOverview from "./ShuttlesOverview";
+import React, { ReactElement, useState, useEffect } from "react";
+import ShuttlesMap from "./ShuttlesMap";
+import StationDropdown from "./StationDropdown";
 import { Route } from "../../__v3api";
+import { Coordinates, Diversion, MaybeDirectionId } from "./__shuttles";
 
 interface Props {
   route: Route;
+  diversions: Diversion;
 }
 
-const ShuttlesPage = ({ route }: Props): ReactElement<HTMLElement> => {
+const ShuttlesPage = ({
+  route,
+  diversions
+}: Props): ReactElement<HTMLElement> => {
   const places =
     route.id === "Green" ? route.direction_names : route.direction_destinations;
 
+  const [selectedDirectionId, setSelectedDirectionId] = useState<
+    MaybeDirectionId
+  >(null);
+  const [displayedShapes, setDisplayedShapes] = useState([
+    ...diversions.shapes
+  ]);
+  const [displayedStops, setDisplayedStops] = useState([...diversions.stops]);
```

Comment:
Is `[...diversions.stops]` not equivalent to `diversions.stops`? (and likewise for `shapes`)

Reply:
Yes! Because in JavaScript arrays are mutable, and in this case I wanted to copy the array (in this case I happen to use the spread operator `...` to achieve that) and not maintain a reference to the original.

Reply:
Ah, right, I missed that the intent here was to copy the array — the way I was familiar with was `array.slice()`.